cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/main.py

```diff
@@ -0,0 +1,459 @@
+# Copyright 2020 Google LLC.
+# This software is provided as-is, without warranty or representation
+# for any use or purpose.
+# Your use of it is subject to your agreement with Google.
+
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Background Cloud Function for loading data from GCS to BigQuery.
+"""
+import json
+import re
+from collections import deque
+from os import getenv
+from pathlib import Path
+from time import monotonic, sleep
+from typing import Any, Deque, Dict, List, Optional, Tuple
+from urllib.parse import urlparse
+
+from google.api_core.client_info import ClientInfo
+from google.api_core.exceptions import PreconditionFailed
+from google.cloud import bigquery, storage
+from google.cloud.exceptions import NotFound
+
+# 15TB per BQ load job.
+DEFAULT_MAX_BATCH_BYTES = str(15 * 10**12)
+MAX_URIS_PER_LOAD = 10**4
```

Comment:
Nit, but perhaps rename to MAX_SOURCE_URIS_PER_LOAD and/or add to above comment the following link: https://cloud.google.com/bigquery/quotas#load_jobs